zcash/src/private_key.rs

```diff
@@ -261,16 +294,22 @@ impl<N: ZcashNetwork> FromStr for ZcashPrivateKey<N> {
         let b58 = s.from_base58();
         let hex = hex::decode(s);
 
-        // Transparent
         if b58.is_ok() && hex.is_err() {
             let data = b58.unwrap();
             if data.len() != 37 && data.len() != 38 {
                 return Err(PrivateKeyError::InvalidByteLength(data.len()));
             }
+
+            // Sprout
+            if &s[..2] == "SK" {
+                return Self::sprout(s);
+            }
+
+            // Transparent
             return Self::p2pkh(s);
```

Comment:
Please verify the address prefix is valid in an if statement. At the end of the section, return Err